Session/Conversations/ConversationVC.swift

```diff
@@ -187,9 +296,50 @@ final class ConversationVC : BaseVC, ConversationViewModelDelegate, OWSConversat
         // Constraints
         view.addSubview(messagesTableView)
         messagesTableView.pin(to: view)
-        view.addSubview(scrollButton)
-        scrollButton.pin(.right, to: .right, of: view, withInset: -16)
-        scrollButtonConstraint = scrollButton.pin(.bottom, to: .bottom, of: view, withInset: -16)
+        
+        // Blocked banner
+        addOrRemoveBlockedBanner()
+        
+        // Message requests view & scroll to bottom
+        view.addSubview(footerControlsStackView)
+        
+        scrollButton.translatesAutoresizingMaskIntoConstraints = false
+        footerControlsStackView.addArrangedSubview(scrollButton)
+        footerControlsStackView.addArrangedSubview(messageRequestView)
+        
+        messageRequestView.addSubview(messageRequestDescriptionLabel)
+        messageRequestView.addSubview(messageRequestAcceptButton)
+        messageRequestView.addSubview(messageRequestDeleteButton)
+        
+        let footerControlsStackViewBottomConstraint: NSLayoutConstraint = footerControlsStackView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -16)
```

Comment:
We have some encapsulated functions in `SessionUIKit` -> `UIView+Constraints` for layout. Could you use those functions instead?

Reply:
I actually really don't like the `UIView+Constraints` extensions (and that there is a bunch of `PureLayout` mixed in elsewhere as well) - by using them we are basically inventing our own constraint management language which increases the learning curve for new people working on the project. And, with the exception of `pin(to:)` and `pin(to:withInset)` it doesn't really save much code (most things still take the same number of lines).

I'd also argue that by using `NSLayoutConstraint.activate()` you incentivise devs to keep all of their layout code together (since adding multiple `NSLayoutConstraint.activate` calls is a tiny inconvenience) whereas with the `pin()` functions seem to be in multiple places in a lot of the files they are used in (which makes it more likely to introduce bugs when modifying layouts since they are easy to miss).

If you take a look at `MessageRequestsViewController`, `MessageRequestsCell` or `MediaMessageView` you can see how the layout code is in a standardised location in each (the first two also have a nice separation between view configuration, view hierarchy setup and layout setup).